src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeBuilderImpl.cs

```diff
@@ -217,6 +240,9 @@ protected override void SetParentCore([DynamicallyAccessedMembers(DynamicallyAcc
         public override string Name => _name;
         public override Type? DeclaringType => _declaringType;
         public override Type? ReflectedType => _declaringType;
+        public override bool IsGenericTypeDefinition => IsGenericType;
+        public override bool IsGenericType => _typeParameters != null;
+        public override Type[] GenericTypeParameters => _typeParameters ?? EmptyTypes;
```

Comment:
Should this return a copy?

Reply:
Hmm, the runtime impl does not return a copy. I would add a comment that says we are not returning a copy for better compat.